dotcms-integration/src/test/java/com/dotmarketing/portlets/categories/business/CategoryFactoryTest.java

```diff
@@ -276,4 +291,271 @@ public void Test_Has_Dependencies() throws DotDataException {
         assertTrue(categoryFactory.hasDependencies(root));
     }
 
+    /**
+     * Method to test: {@link CategoryFactoryImpl#findAll(CategoryFactory.CategorySearchCriteria)}
+     * When:
+     *
+     * - Create a random string to be used as the filter for the test.
+     * - Create two top-level categories, named topLevelCategory_1 and topLevelCategory_2, and include the filter in their names.
+     * - For topLevelCategory_1, create four children:
+     *     Include the filter in three of these children: one in the key, one in the name, and one in the variable name.
+     *     The fourth child should not include the filter anywhere.
+     * - Add a child to the last child of topLevelCategory_1 (the one without the filter) and include the filter in its name.
+     * Also, create a grandchild and include the filter in its name.
+     * - Create another child for topLevelCategory_1 and include the filter in the key, name, and variable name.
+     * - Call the method with the filter in both lowercase and uppercase.
+     *
+     * Should:
+     *
+     * Return five categories: the three children of topLevelCategory_1 that include the filter and the two grandchildren.
+     */
+    @Test
+    public void getAllCategoriesFiltered() throws DotDataException {
+
+        final String stringToFilterBy = new RandomString().nextString();
+
+        final Category topLevelCategory_1 = new CategoryDataGen().setCategoryName("Top Level Category " + stringToFilterBy)
```

Comment:
Let's also include a test case for when the "stringToFilterBy" is in the "middle" (surrounded by other strings both at the beginning and at the end, this to test the %filter%)

Reply:
done https://github.com/dotCMS/core/pull/29021/commits/15aa5848cb228fd92ea4d68fe0f1580e8a301065#diff-ca0b19817968af90ad2ada127d0180e9d48d83bde46ba6dd038ed2f9bfc11aacR368-R379